src/plugins/techStackGeneric.nim

```diff
@@ -123,20 +123,21 @@ proc scanFile(filePath: string, category: string, subcategory: string) =
     scanFileStream(stream)
 
 proc getProcNames(): HashSet[string] =
+  ## Returns every Name value in files at `/proc/[0-9]+/status`.
+  ## This is the filename of each executable, truncated to 15 characters.
   result = initHashSet[string]()
   for kind, path in walkDir("/proc/"):
-    for ch in path.splitPath().tail:
-      try:
-        if ch notin "0123456789":
+    if kind == pcDir and path.lastPathPart().allIt(it in {'0'..'9'}):
+      let data =
+        try:
+          readFile(path / "status")
+        except:
           continue
```

Comment:
We have a utility in nimutils tryToLoadFile. Let's switch to that so chalk is consistent with file reads. Once the fd_cache will move to nimutils we can adjust it to use the cache as well